src/traversals/dfs.jl

```diff
@@ -32,28 +32,30 @@ function is_cyclic end
 end
 # see https://github.com/mauro3/SimpleTraits.jl/issues/47#issuecomment-327880153 for syntax
 @traitfn function is_cyclic(g::AG::IsDirected) where {T,AG<:AbstractGraph{T}}
+    # 0 if not visited, 1 if visited, 2 if in the current dfs path, 3 if fully explored 
     vcolor = zeros(UInt8, nv(g))
+    vertex_queue = Vector{T}()
```

Comment:
Why a Vector instead of another data structure?

Reply:
I kept the last one, why do you want something else ? We only need to push and pop, there are no weight ordering

Reply:
There should be some queue data structures in [DataStructures.jl](https://github.com/JuliaCollections/DataStructures.jl) - but I am not sure if and how much more efficient they would be. But if I understand correctly, you always add or remove at the end, so it should be called `stack` not `queue`.

Reply:
I think we should merge this as is (with a renaming) and keep track of the potential for a better stack in a new issue?